docroot/modules/humsci/hs_config_readonly/src/EventSubscriber/ConfigReadOnlyEventSubscriber.php

```diff
@@ -0,0 +1,222 @@
+    // entity.
+    if (!$mark_form_read_only && $form_object instanceof EntityFormInterface) {
+      $entity = $form_object->getEntity();
+      $mark_form_read_only = $entity instanceof ConfigEntityInterface;
+    }
+
+    // Don't block particular patterns.
+    if ($mark_form_read_only && $form_object instanceof EntityFormInterface) {
+      $entity = $form_object->getEntity();
+      $name = $entity->getConfigDependencyName();
+
+      // Block config from a module.
+      $mark_form_read_only = $this->configIsEditable([$name]);
+
+      // Unless its defined as a whiteist pattern.
+      if ($this->matchesWhitelistPattern($name)) {
+        $mark_form_read_only = FALSE;
+      }
+    }
+
+    // Config forms.
+    if ($mark_form_read_only && $form_object instanceof ConfigFormBase) {
+      $editable_config = $this->getEditableConfigNames($form_object);
+      $mark_form_read_only = $this->configIsEditable($editable_config);
+    }
+
+    if ($mark_form_read_only) {
+      $event->markFormReadOnly();
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    $events = [];
+    $events[ReadOnlyFormEvent::NAME][] = ['onFormAlter', 200];
+    return $events;
+  }
+
+  /**
+   * Get the editable configuration names.
+   *
+   * @param ConfigFormBase $form
+   *   The configuration form.
+   *
+   * @return array
+   *   An array of configuration object names that are editable if called in
+   *   conjunction with the trait's config() method.
+   *
+   * @throws \ReflectionException
+   */
+  protected function getEditableConfigNames(ConfigFormBase $form) {
+    // Use reflection to work around getEditableConfigNames() as protected.
+    // @todo Review in 9.x for API change.
+    // @see https://www.drupal.org/node/2095289
+    $reflection = new \ReflectionMethod(get_class($form), 'getEditableConfigNames');
+    $reflection->setAccessible(TRUE);
+    return $reflection->invoke($form);
+  }
+
+  /**
+   * Check if the given configuration name can be edited.
+   *
+   * @param array $config
+   *   Array of config names.
+   *
+   * @return bool
+   *   If it can be edited.
+   */
+  protected function configIsEditable(array $config) {
+    if ($config == array_filter($config, [$this, 'isEditableConfig'])) {
+      return FALSE;
+    }
+    return TRUE;
+  }
+
+  /**
+   * @param $name
+   *
+   * @return bool
+   */
+  protected function isEditableConfig($name) {
+    foreach ($this->getLockedConfigs() as $config) {
+      if ($config == $name) {
+        return FALSE;
+      }
+    }
+    return TRUE;
+  }
+
+  /**
+   * Get all configs provided by modules.
+   *
+   * @return array
+   *   Config names.
+   */
+  protected function getLockedConfigs() {
+    $module_configs = [];
+    foreach ($this->moduleHandler->getModuleList() as $name => $extension) {
+      if ($extension->getType() == 'profile' || in_array($name, $this->excludedModules)) {
```

Comment:
Why not profiles?

Reply:
I thought maybe profiles would be a starting point. i kinda went both directions with that.

Reply:
This seems like something that should be configurable. The peel-and-stick approach to this module works but what if this was based off of distribution?

Reply:
I made it configurable. `hs_config_readonly.yml` can define which modules to ignore. by default if the config doesn't exist, it ignores `system` module.

Reply:
Should we make a UI for this?

Reply:
or it can be easily changed via Drush on a per-site basis but be hidden from the user.